src/world_interface/data.h

```diff
@@ -233,3 +233,7 @@ class LimitSwitchData {
 };
 
 } // namespace robot::types
+
+namespace std {
+std::string to_string(const robot::types::CameraID& id);
+}
```

Comment:
Good catch, I actually didn't know that - in my protocol branch I did it for a different type, so I should probably change that. Maybe we could make an overloaded cast operator instead? I'll have to see if something like that compiles

Reply:
Would it be a good idea to add a templated `util::to_string()` method? That way, the un-specialized implementation can just invoke `std::to_string()` and then for custom types we can provide a template specialization. That way, client code can just always call `util::to_string()`, regardless of the type.

What's nice is the template specialization can be anywhere since it doesn't have to be in a header file, so users don't have to include a separate function prototype based on the type they want to convert to a string; they can just include the file with `util::to_string()` and as long as the specialization gets compiled then they're good.

I don't think nonmember cast operators are allowed, so this might be a good idea. Does that make sense, or do you see a better solution?

Reply:
I think that would be good - it would give us something we could extend ourselves without problems, but still work as a fallback if we choose not to extend it. I'll implement this asap and check it in

Reply:
Sorry this took so long - been a lot busier than expected with work and job interviews and stuff. I made a templated `util::to_string()` method and then a specialization for `CameraID`. In my other branch I will change the overloaded `std::to_string` to a specialization of `util::to_string` as well.